worksheet_D/oxo.py

```diff
@@ -1,26 +1,64 @@
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        self.gameboard = []
+        for i in xrange(9):
+            self.gameboard.append(0)
+
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        square = x + y * 3
+        return self.gameboard[square]
+
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        square = x + y * 3
+        if self.gameboard[square] == 0:
+            self.gameboard[square] = mark
+            return True
+        else:
+            return False
+
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        for squares in xrange(9):
+            if self.gameboard[squares] == 0:
+                return False
+        return True
+
 
     def get_winner(self):
         """ If a player has three in a row, return 1 or 2 depending on which player.
             Otherwise, return 0. """
-        raise NotImplementedError("TODO: implement get_winner")
+
+        for i in xrange(3):
+
+            # Checks for vertical lines of 3
+            # When i = 1, i + 3 = 4 and i + 6 = 7
+            # 1, 4, 7 is a vertical line on the grid
+            if self.gameboard[i] == self.gameboard[i + 3] == self.gameboard[i + 6] != 0:
```

Comment:
OK, but could be made more readable by using `self.get_square()` rather than accessing `self.gameboard` directly